lib/Persistence/AbstractPersistence.php

```diff
@@ -82,21 +82,23 @@ protected static function _exists($filename)
      */
     protected static function _initialize()
     {
-        // Create storage directory if it does not exist.
-        if (!is_dir(self::$_path)) {
-            if (!@mkdir(self::$_path, 0700)) {
-                throw new Exception('unable to create directory ' . self::$_path, 10);
+        if (property_exists($data->meta, 'webserver') && $data->meta->webserver && $this->_conf->getKey('webserver') == "Apache") {
```

Comment:
If you have a better idea, just explain it…
Comments are there to discuss things. IMHO "autodetection" as it was (partly) implemented in your first PR is a good idea.